modules/eslint-plugin/spec/rules/component-store/avoid-mapping-component-store-selectors.spec.ts

```diff
@@ -0,0 +1,141 @@
+import type {
+  ESLintUtils,
+  TSESLint,
+} from '@typescript-eslint/experimental-utils';
+import { fromFixture } from 'eslint-etc';
+import * as path from 'path';
+import rule, {
+  messageId,
+} from '../../../src/rules/component-store/avoid-mapping-component-store-selectors';
+import { ruleTester } from '../../utils';
+
+type MessageIds = ESLintUtils.InferMessageIdsTypeFromRule<typeof rule>;
+type Options = ESLintUtils.InferOptionsTypeFromRule<typeof rule>;
+type RunTests = TSESLint.RunTests<MessageIds, Options>;
+
+const validConstructor: () => RunTests['valid'] = () => [
+  `
+import { ComponentStore } from '@ngrx/component-store'
+
+class Ok extends ComponentStore<MoviesState> {
+  movies$ = this.select((state) => state.movies);
+  firstMovie$ = this.select(this.movies$, (movies) => movies[0]);
+
+  constructor() {
+    super({ movies: [] })
+  }
+}`,
+  `
+import { ComponentStore } from '@ngrx/component-store'
+
+class Ok {
```

Comment:
I saw in other test cases that we used to add incremental number value to each Ok/NotOk cases. I am not sure whether is somehow required.

Reply:
It's not required, I didn't add them here because other test cases weren't consistent and it can sometimes take some times to increment the index while adding tests 😅